internal/app/mistapiconnector/mistapiconnector_app.go

```diff
@@ -473,14 +484,41 @@ func (mc *mac) generateRouteKeys(stream *livepeer.CreateStreamResp) []string {
 	return keys
 }
 
+func (mc *mac) sign(sharedSecret string) string {
+	s := sha256.New()
+	s.Write([]byte(sharedSecret))
+	signature := hex.EncodeToString(s.Sum(nil))
+	return signature
+}
+
 func (mc *mac) triggerUserNew(w http.ResponseWriter, r *http.Request, lines []string, rawRequest string) bool {
+
 	if len(lines) != 6 {
 		glog.Errorf("Expected 6 lines, got %d, request \n%s", len(lines), rawRequest)
 		w.WriteHeader(http.StatusBadRequest)
 		w.Write([]byte("false"))
 		return false
 	}
 
+	u := &userNewPayload{
+		StreamName:    	lines[0],
+		IpAddress:     	lines[1],
+		UserAgent:     	lines[2],
+		OutputProtocol: lines[3],
+		RequestUrl:    	lines[4],
+		SessionID:     	lines[5],
```

Comment:
Hm, how come does `mist` have the stream name and session IDs nicely sent in the response like this? Are they Mist abstractions instead? If so, we should be interpreting all these and only send values that make sense to API users here (the "connecting" part of mist-api-connector 😆)

So for example, I already see that `lines[0]` is also used below as the playback ID. So it certainly does not mean the "Stream Name" to an API user. Nor do I think mist has any notion of our "session ID" in the API, that stays within `MistProcLivepeer` logic. 

So... You can only actually build this payload after you get the `*streamInfo` from the `pub2info` map. From it you can populate this payload with both some mist-specific data that make sense to API users like IP address, user agent, request URL, playback/stream ID, webhook ID, event ID etc.

Reply:
Also, realized that the webhook payload should actually follow the format from our other webhooks, only changing what is inside the payload. This is where we create webhooks in the API: https://github.com/livepeer/livepeer-com/blob/6b86fb011c94e9c6dee5c9837110e0bdf7f08663/packages/api/src/webhooks/cannon.ts#L384-L392

So you should have all these fields: id, webhookId, createdAt, timestamp, event, stream (which I guess we can leave empty or only have an `{id: string}` field) and then a `payload` field where you can have the extra information specific to this webhook (you shouldn't need to repeat any info if it's already in the outer schema either, like the info about the stream)